Tests/SmokeDynamoDBTests/DynamoDBCompositePrimaryKeyTableUpdateItemConditionallyAtKeyTests.swift

```diff
@@ -44,9 +44,29 @@ class DynamoDBCompositePrimaryKeyTableUpdateItemConditionallyAtKeyTests: XCTestC
     
     func getUpdatedPayloadProviderAsync(on eventLoop: EventLoop) -> ((TestTypeA) -> EventLoopFuture<TestTypeA>) {
         func provider(item: TestTypeA) -> EventLoopFuture<TestTypeA> {
-            let promise = eventLoop.makePromise(of: TestTypeA.self)
-            promise.succeed(TestTypeA(firstly: "firstlyX2", secondly: "secondlyX2"))
-            return promise.futureResult
+            return eventLoop.makeSucceededFuture(TestTypeA(firstly: "firstlyX2", secondly: "secondlyX2"))
+        }
+        
+        return provider
+    }
+    
+    typealias TestTypeADatabaseItem = StandardTypedDatabaseItem<TestTypeA>
+    func updatedItemProvider(item: TestTypeADatabaseItem) -> TestTypeADatabaseItem {
+        let key = StandardCompositePrimaryKey(partitionKey: "partitionId",
+                                              sortKey: "sortId")
+        return TestTypeADatabaseItem.newItem(
+            withKey: key,
+            andValue: TestTypeA(firstly: "firstlyX2", secondly: "secondlyX2"))
+    }
+    
+    func getUpdatedItemProviderAsync(on eventLoop: EventLoop) -> ((TestTypeADatabaseItem) -> EventLoopFuture<TestTypeADatabaseItem>) {
```

Comment:
Should we rename this to make it clearer that the updated item provided returned by the function is async in terms of EventLoopFuture, not async/await?

Reply:
Fixed